### PR TITLE
Update deleteTag to be case-insensitive

### LINKED ISSUE #156: [PE-D] Try to delete tag but failed
Task one has tag 'Hi'. Try to delete the tag but failed and the error shown is 'the tag doesn't exist'

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: austenjs/ped#9

### LINKED ISSUE #229: Allow deleteTag to delete case-insensitive tags

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
Fixes #156, fixes #229

### CODE REVIEW

**Review by QY-H00 — COMMENTED** [review]
Overall LGTM. I may test it after merging. Well done bro!

File: src/main/java/seedu/module/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -54,18 +54,18 @@ public static Name parseName(String name) throws ParseException {
...
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code deadline} is invalid.
+     * @throws ParseException if the given {@code time} is invalid.
      */
     public static Time parseTime(String deadline) throws ParseException {
         requireNonNull(deadline);
-        String trimmedDeadline = deadline.trim();
-        if (!Time.isValidTime(trimmedDeadline)) {
+        String trimmedTime = deadline.trim();
```
Comment (umergta) [inline src/main/java/seedu/module/logic/parser/ParserUtil.java:64]: good catch!

File: src/test/java/seedu/module/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -84,46 +84,46 @@ public void parseName_validValueWithWhitespace_returnsTrimmedName() throws Excep
...
-        Time expectedDeadline = new Time(VALID_DEADLINE);
-        assertEquals(expectedDeadline, ParserUtil.parseTime(deadlineWithWhitespace));
+    public void parseTime_validValueWithWhitespace_returnsTrimmedTime() throws Exception {
+        String timeWithWhitespace = WHITESPACE + VALID_TIME + WHITESPACE;
+        Time expectedTime = new Time(VALID_TIME);
+        assertEquals(expectedTime, ParserUtil.parseTime(timeWithWhitespace));
     }
 
     @Test
-    public void parseAddress_null_throwsNullPointerException() {
```
Comment (umergta) [inline src/test/java/seedu/module/logic/parser/ParserUtilTest.java:110]: oh damn another nice catch HAHA

**Review by umergta — COMMENTED** [review]
LGTM. Nice work on catching some wrong method and variable names!

**Pull request merged**